### DISCUSSION TITLE
Frequency vs misorientation distribution plot

### DISCUSSION THREAD

**Discussion by huseyinhzl** [mtex-toolbox/mtex#2216] [Ask Anything]
Hello,

I am working on Ti6Al4V alloys and I need to plot pixel-to-pixel misorientation frequency distribution for my parent beta grains to show a possible twinning in the sample. However, I could not find how to do that in the documentation, so wanted to ask help. Anyone know how I can plot pixel-to-pixel misorientation frequency distribution?

Thank you

**Comment by zmichels** [comment]
A while back, when I was trying to get a complete set of neighbor-pair misorientations, I worked out a way to utilize some of the built-in MTEX code. To do this I dug into the KAM functions iirc (as it is quite efficient), and I saw it used a couple functions (something like: spatialDecomposition and UnitCell) that could identify and subset neighbor pairs of orientations for processing.
I put an example (probably requires some tweaking for any given project / application) on GitHub here:
https://github.com/zmichels/misorientations

If you can share an example of your typical datasets, I can cobble together an example of how I would use some of the scripts/functions in that repo to get what you want. I do this often... for example to make misorientation angle histograms of *all* pairs of a phase. It's also nice to then spatially identify parts of the map and/or grain interiors with some misorientation range/threshold. I find things like subgrain boundaries often present over multiple pixels, rather than a discrete boundary, such as might be fit by grains.innerBoundary.

Zach

**Reply by zmichels** [reply]
This is the main script to look at I think if you check it out:

https://github.com/zmichels/misorientations/blob/master/neighborMisOrs.m

**Reply by huseyinhzl** [reply]
Hi Zach,

Thank you for your reply. I will check the repository and try to get the idea. This pixel-to-pixel plot is essential for me to prove a phenomena in my thesis but could not get it done. 

I do not mind share my data with you, but my .ctf file is quite a big map (around 5M pixels data) and I am struggling even to do parent grain reconstruction, so I am also trying to find a solution for big data issue. If you still want to have a look at the data, I can share it with you.

**Reply by zmichels** [reply]
Sounds great -- I'd enjoy to take a look.

**Reply by huseyinhzl** [reply]
then I will need your mail address since I cannot add the file here due to its size :)

**Reply by zmichels** [reply]
Sure!
zachary.michels@gmail.edu

**Reply by zmichels** [reply]
If it's too big to send/email. You can upload it to a folder I made here:

https://drive.google.com/drive/folders/1PCVaSVCLoMeuVD7Cos7dmibi659TvJ-R?usp=sharing

**Reply by huseyinhzl** [reply]
I emailed but also uploading to the folder you have created just in case

**Reply by zmichels** [reply]
Got it and putting something together now.

**Reply by zmichels** [reply]
I uploaded some more files to that folder I made:(https://drive.google.com/drive/folders/1PCVaSVCLoMeuVD7Cos7dmibi659TvJ-R?usp=drive_link)

The folder contains an example script I wrote and a histogram result from using the script with a wide range of threshold angles (0.5–90). LMK if I can help with any of it.

Cheers,
Zach


![P100_PD30_ET300_sub1_histogram](https://github.com/user-attachments/assets/37dc4c87-fa77-49f4-a981-7ffc2c81ef56)

![P100_PD30_ET300_sub1_Axes_spec](https://github.com/user-attachments/assets/aac2c625-649e-4780-8373-de3c1de1646f)
![P100_PD30_ET300_sub1_Axes_xtal](https://github.com/user-attachments/assets/be215232-7237-433a-934f-dfe7448ebaf3)

**Reply by huseyinhzl** [reply]
Hi Zach,

Sorry for missing these. They look great but I think the misorientation distribution is for the alpha phase. I am interested in BCC beta phase which EBSD does not index. My question is if I can manage to post-process that big data for parent grain reconstruction can I use the code that you have shared? Also, do you have any idea how I can run the parent grain reconstruction in smaller regions of the large area map? I have tried to run couple of codes but I kept failing or corrupting the file.

**Reply by zmichels** [reply]
The code I shared is really only for misorientations, but can be pointed at any phase of interest to try. I am no good for help on phase-transformative reconstruction... but others in forum are indeed experts.
